### PR TITLE
More codelist validation

### PR DESCRIPTION
**Pull request by DanielElisenberg**
* Added extra testing for codelists to ensure that no duplicate code definitions overlap each other in the same time span
* Added tests for unit_type_variables where they are fully tested in the context of a generated metadata file (including the new code list checks)
* Updated metadata code list errors for KOMMUNE. I would like it if you could take a look at the changes in KOMMUNE.json @JohanSjoberg99

### CODE REVIEW

File: microdata_tools/validation/components/unit_type_variables/KOMMUNE.json [hunk 1]
```diff
@@ -10412,17 +10368,6 @@
...
-        "code": "5412",
-        "categoryTitle": [
-          {
-            "languageCode": "no",
-            "value": "Dielddanuorri - Tjeldsund"
-          }
-        ],
-        "validFrom": "2020-01-01",
-        "validUntil": null
-      },
```
Comment (JohanSjoberg99) [inline microdata_tools/validation/components/unit_type_variables/KOMMUNE.json:10425]: According to KLASS. This is the correct one: 5412 - Dielddanuorri - Tjeldsund. @DanielElisenberg

File: microdata_tools/validation/components/unit_type_variables/KOMMUNE.json [hunk 2]
```diff
@@ -11017,17 +10962,6 @@
...
-        "code": "2100",
-        "categoryTitle": [
-          {
-            "languageCode": "no",
-            "value": "Ukjent kommune i Svalbard"
-          }
-        ],
-        "validFrom": "1919-01-01",
-        "validUntil": "2019-01-01"
-      },
```
Comment (JohanSjoberg99) [inline microdata_tools/validation/components/unit_type_variables/KOMMUNE.json:11030]: Will check to see if we should keep "ukjent kommune i Svalbard" and delete the other one.

**Review by JohanSjoberg99 — CHANGES_REQUESTED** [review]
Keep 5412 - Dielddanuorri - Tjeldsund, remove 4512 - Tjeldsund.
Will check if "ukjent komune" or just "kommune

**Review by pawbu — APPROVED** [review]
Well done!

**Review by JohanSjoberg99 — APPROVED** [review]
All changes is according to agreement.

**Review by pawbu — APPROVED** [review]
💯
